SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -101,17 +101,32 @@ void ComponentReader::ApplyComponentData(const Worker_ComponentData& ComponentDa
 
 	Schema_Object* ComponentObject = Schema_GetComponentDataFields(ComponentData.schema_type);
 
-	TArray<uint32> UpdatedIds;
-	UpdatedIds.SetNumUninitialized(Schema_GetUniqueFieldIdCount(ComponentObject));
-	Schema_GetUniqueFieldIds(ComponentObject, UpdatedIds.GetData());
+	// ComponentData will be missing fields if they are completely empty (options, lists, and maps).
+	// However, we still want to apply this empty data, so we need to reconstruct the full
+	// list of field IDs for that component type (Data, OwnerOnly).
+	const TArray<Schema_FieldId>& InitialIds = ClassInfoManager->GetFieldIdsByComponentId(ComponentData.component_id, Channel.GetObjectRepLayout(&Object));
```

Comment:
So this `Channel.GetObjectRepLayout(&Object)` has a check that the replicator exists inside. The good news is that replicators are created on ActorChannel creation, so we should be good. They can be skipped in native, but I don't think we ever skip creating them in Spatial.

Reply:
This seems brittle, we are relying on an interplay between how we deal with actor channels and where we are likely to need this information. Is there a way to remove the dependency? Like adding this data to UHT or to some asset?

Reply:
Ultimately all we need is the cleared fields (although it might be useful to have all of them)

Reply:
True, the only nice way I can think of, is to store this info in the schema database. I did not want to mess with it this time around, but maybe I can create a follow-up ticket.

Reply:
Also, it's slightly less brittle than it seems, I think, there are a lot of checks around this (runtime, true, but still)

Reply:
Please do create the follow up ticket. I suspect it won't be done any time soon, but that is the correct place for this to live. Right now this relies on hidden assumptions and we have a weird check.